src/PubSubClient.cpp

```diff
@@ -586,3 +618,13 @@ PubSubClient& PubSubClient::setStream(Stream& stream){
 int PubSubClient::state() {
     return this->_state;
 }
+
+boolean PubSubClient::setBufferSize(uint16_t size) {
+  this->buffer = (uint8_t*)realloc(this->buffer, size);
```

Comment:
This should use a temp variable. If realloc fails it will return a NULL pointer without freeing the original memory. Thus, if NULL is returned this method should not make any change to this->buffer since it is still valid.

Reply:
yes. this makes sense.

@knolleary is this pull request likely to ever be merged? is it worth me doing this?